runtime/major_gc.c

```diff
@@ -1150,14 +1159,13 @@ static void cycle_all_domains_callback(caml_domain_state* domain, void* unused,
 
       atomic_store(&domain_global_roots_started, WORK_UNSTARTED);
 
-      /* Reset GC work accounting counters. */
-      for (i = 0; i < Max_domains; i++){
-        domain_accounts[i].alloc = 0;
-        domain_accounts[i].dependent = 0;
-        domain_accounts[i].extra = 0.0;
-      }
+      /* reset work counters */
+      caml_plat_lock (&accounting_lock);
+      caml_enumerate_participating_domains (&reset_domain_account);
```

Comment:
I'm confused, aren't the arguments `int participating_count, caml_domain_state **participating` precisely supposed to let us do this (iterate on all participating domains)? If it is enough to access the `stw_domains` globals, why do we have them in the STW API?